functions/index.js

```diff
@@ -1,43 +1,32 @@
 const functions = require('firebase-functions');
 const Discord = require('discord.js');
-const axios = require('axios');
+const { handleDailyCC } = require("./dailyCC")
 const admin = require('firebase-admin');
-admin.initializeApp();
-const slugs = require('./constants');
+admin.initializeApp({
+  credential: admin.credential.applicationDefault()
+});
 
-exports.dailycc = functions.https.onRequest(() => {
-	const client = new Discord.Client();
+exports.dailycc = functions.https.onRequest(async (req, res) => {
+	const discordClient = new Discord.Client();
 	const { access_token } = functions.config().discord;
+	const firestore = admin.firestore();
 
-	client.login(access_token).catch((err) => console.log(err));
-
-	client.on('ready', () => {
-		const slug = slugs[Math.round(Math.random() * slugs.length - 1)];
-		postQuestion(slug, client.channels.cache.get('771821709589479505'));
-	});
+	discordClient.login(access_token).catch((err) => console.log(err));
 
-	const postQuestion = async (slug, channel) => {
-		const result = await getQuestion(slug);
+	discordClient.on('ready', async () => {
+		const channelId = "771821709589479505"
+		const channel = discordClient.channels.cache.get(channelId);
 
-		if (result === 'Not Found') {
-			channel.send(`No coding problem today. Check back tomorrow 😁`);
-			return;
-		} else {
-			channel.send(result).catch((err) => {
-				channel.send('There was an issue grabbing the question');
-				channel.send(`Error: ${err.message}`);
-			});
+		let question
+		try{
+			question = await handleDailyCC({channel, firestore})
```

Comment:
Merge lines 20 and 22 so that it's in the try and remove firestore since it'll be in its own config.